plugins/csp-simple-bodies/src/SimpleBody.cpp

```diff
@@ -211,22 +170,45 @@ float getRingShadow() {
   #endif
 }
 
+// placeholder for the BRDF in HDR mode
+$BRDF_HDR
+
+// placeholder for the BRDF in light mode
+$BRDF_NON_HDR
+
 void main() {
     oColor = texture(uSurfaceTexture, vTexCoords).rgb;
 
-    #ifdef ENABLE_HDR
-      // Make the amount of ambient brightness perceptually linear in HDR mode.
-      float ambient = pow(uAmbientBrightness, E);
-      oColor = SRGBtoLINEAR(oColor) * uSunIlluminance / PI;
-    #else
-      float ambient = uAmbientBrightness;
-      oColor = oColor * uSunIlluminance;
-    #endif
-
-    #ifdef ENABLE_LIGHTING
-      vec3 light = getRingShadow() * getEclipseShadow(vPosition) * orenNayar(normalize(vNormal), normalize(vSunDirection), -normalize(vPosition));
-      oColor = mix(oColor * light, oColor, ambient);
-    #endif
+    // Needed for the BRDFs.
+    vec3 N = normalize(vNormal);
+    vec3 L = normalize(vSunDirection);
+    vec3 V = normalize(-vPosition);
+    float cos_i = dot(N, L);
+    float cos_r = dot(N, V);
+
+#ifdef ENABLE_HDR
+    // Make the amount of ambient brightness perceptually linear in HDR mode.
+    float ambient = pow(uAmbientBrightness, E);
+    oColor = SRGBtoLINEAR(oColor) * uSunIlluminance / $AVG_LINEAR_IMG_INTENSITY;
+    float f_r = BRDF_HDR(N, L, V);
+#else
+    float ambient = uAmbientBrightness;
+    oColor = oColor * uSunIlluminance;
```

Comment:
This line works because `uSunIlluminance` seems to be 1 here always. I couldn't figure out quickly if it serves some purpose, so just marking it.